packages/pricing/src/services/rule-type.ts

```diff
@@ -70,6 +71,8 @@ export default class RuleTypeService<TEntity extends RuleType = RuleType> {
     data: PricingTypes.CreateRuleTypeDTO[],
     @MedusaContext() sharedContext: Context = {}
   ): Promise<TEntity[]> {
+    validateRuleAttributes(data.map((d) => d.rule_attribute))
```

Comment:
thought(non-blocking): Should this live at the workflow or Rest API level instead? 

This is slightly opinionated business logic. If someone wants to replace the data layer, these attributes will remain forbidden.

I realize this is a farfetched example. However, think it's important to draw strict lines between what lives at the workflow level vs. the service level.

That said, maybe this is an exception to the rule because the validation relates to columns in the database.

Wdyt?

Reply:
or maybe it could be done in the repository layer?

Reply:
I think this makes sense for it to be in the module more than the core, since this is a validation for the module logic, but wouldn't mind moving this over to a repository level.

I would like to learn more about use-cases that you guys have envisioned. What does replacing the data layer here mean?

1. if the ORM is going to be changed, it would still make sense for this logic to be present. 
2. if the calculatePricing logic is entirely different, this validation wouldn't even be needed.